contracts/Market.sol

```diff
@@ -1066,23 +1074,49 @@ contract Market is
    * @return The amount of the fee for Nori.
    */
   function calculateNoriFee(uint256 amount) external view returns (uint256) {
-    return amount.mulDiv(_priceMultiple * _noriFeePercentage, 10000);
+    return
+      convertRemovalAmountToPurchasingTokenAmount(
+        amount.mulDiv(_priceMultiple * _noriFeePercentage, 10000)
+      );
+  }
+
+  function convertRemovalAmountToPurchasingTokenAmount(uint256 removalAmount)
+    public
+    view
+    returns (uint256)
+  {
+    if (_purchasingTokenDecimals == 18) {
+      return removalAmount;
+    }
+    int8 decimalDelta = 18 - int8(_purchasingTokenDecimals);
```

Comment:
is there a specific reason for converting this value to a uint8?  gas savings for smaller data types only apply when these things are actually being stored in storage memory (and you can pack them tightly) and it actually can cost more gas to cut something down to a uint8 when handling in-memory since the EVM word size is 256 bits

Reply:
oh I see, it's because the `_purchasingTokenDecimals` variable is a uint8.  I believe that unless this value is part of a struct or an array and is being packed tightly with other values that are sub-256 bytes, there is still no savings on making this a smaller data type than 256 even though it is a storage variable

Reply:
It was mostly doing it to be consisent with the return value of `decimals()` but happy to change it if you think it improves gas.